pkg/kubelet/cm/deviceplugin/pod_devices_test.go

```diff
@@ -0,0 +1,265 @@
+				},
+			},
+		},
+		"get-uid-3": {
+			podUID:   "UID-3",
+			contName: "cont3",
+			expected: &DeviceRunContainerOptions{
+				Envs: []container.EnvVar{
+					{Name: "thirdenvkey3", Value: "thirdenvvalue3"},
+					{Name: "thirdenvkey4", Value: "thirdenvvalue4"},
+					{Name: "thirdenvkey0", Value: "thirdenvvalue0"},
+					{Name: "thirdenvkey1", Value: "thirdenvvalue1"},
+					{Name: "thirdenvkey2", Value: "thirdenvvalue2"},
+				},
+				Mounts: []container.Mount{
+					{Name: "thirdcontainerpath0", ContainerPath: "thirdcontainerpath0", HostPath: "thirdhostpath0", ReadOnly: false, SELinuxRelabel: false, Propagation: 0},
+					{Name: "thirdcontainerpath1", ContainerPath: "thirdcontainerpath1", HostPath: "thirdhostpath1", ReadOnly: false, SELinuxRelabel: false, Propagation: 0},
+					{Name: "thirdcontainerpath2", ContainerPath: "thirdcontainerpath2", HostPath: "thirdhostpath2", ReadOnly: false, SELinuxRelabel: false, Propagation: 0},
+					{Name: "thirdcontainerpath3", ContainerPath: "thirdcontainerpath3", HostPath: "thirdhostpath3", ReadOnly: false, SELinuxRelabel: false, Propagation: 0},
+					{Name: "thirdcontainerpath4", ContainerPath: "thirdcontainerpath4", HostPath: "thirdhostpath4", ReadOnly: false, SELinuxRelabel: false, Propagation: 0},
+				},
+				Devices: []container.DeviceInfo{
+					{PathOnHost: "thirdhostpath0", PathInContainer: "thirdcontainerpath0", Permissions: "thirdpermissions0"},
+					{PathOnHost: "thirdhostpath1", PathInContainer: "thirdcontainerpath1", Permissions: "thirdpermissions1"},
+					{PathOnHost: "thirdhostpath2", PathInContainer: "thirdcontainerpath2", Permissions: "thirdpermissions2"},
+					{PathOnHost: "thirdhostpath3", PathInContainer: "thirdcontainerpath3", Permissions: "thirdpermissions3"},
+					{PathOnHost: "thirdhostpath4", PathInContainer: "thirdcontainerpath4", Permissions: "thirdpermissions4"},
+				},
+			},
+		},
+		"get-none-wrong": {
+			podUID:   "gakki",
+			contName: "masami",
+			expected: nil,
+		},
+		"get-mix-wrong": {
+			podUID:   "UID-3",
+			contName: "cont1",
+			expected: nil,
+		},
+		"get-nil-wrong": {
+			podUID:   "",
+			contName: "",
+			expected: nil,
+		},
+	}
+
+	testPodDevices := getPodDevices()
+	for testName, testCase := range testCases {
+		actual := testPodDevices.deviceRunContainerOptions(testCase.podUID, testCase.contName)
+		if actual == nil || testCase.expected == nil {
+			if actual != testCase.expected {
+				t.Errorf("unexpected result, test: %v", testName)
+			}
+			continue
+		}
+
+		if !reflect.DeepEqual(actual.Mounts, testCase.expected.Mounts) {
+			t.Errorf("unexpected result, test: %v, Mounts not as expected", testName)
+		}
+		if !reflect.DeepEqual(actual.Devices, testCase.expected.Devices) {
+			t.Errorf("unexpected result, test: %v, Devices not as expected", testName)
+		}
+		if !envEqual(actual.Envs, testCase.expected.Envs) {
+			t.Errorf("unexpected result, test: %v, Envs not as expected", testName)
+		}
+
+	}
+}
+
+type sortInterface struct {
```

Comment:
No need for such interface.

 Use below inline code to help get sorted `[]container.EnvVar`
```
sort.Slice(actual.Envs, func(i, j int) bool {
      return actual.Envs[i].Name < actual.Envs[j].Name
})
```

The same for `testCase.expected.Envs`.

Reply:
OK